design/aws-ecs-autoscaling-queue-worker.md

```diff
@@ -0,0 +1,216 @@
+
+  /**
+   * The CMD value to pass to the container. A string with commands delimited by commas.
+   *
+   * @default none
+   */
+  readonly command?: string;
+
+  /**
+   * Number of desired copies of running tasks
+   *
+   * @default 1
+   */
+  readonly desiredTaskCount?: number;
+
+  /**
+   * Flag to indicate whether to enable logging
+   *
+   * @default true
+   */
+  readonly enableLogging?: boolean;
+
+  /**
+   * The environment variables to pass to the container.
+   *
+   * @default none
+   */
+  readonly environment?: { [key: string]: string };
+
+  /**
+   * A name for the queue.
+   *
+   * If specified and this is a FIFO queue, must end in the string '.fifo'.
+   *
+   * @default CloudFormation-generated name
+   */
+  readonly queueName?: string;
+
+  /**
+   * Maximum capacity to scale to.
+   *
+   * @default (desiredTaskCount * 2)
+   */
+  readonly maxScalingCapacity?: number
+}
+```
+
+The `QueueWorkerService` constructs (for EC2 and Fargate) will use the following existing constructs:
+
+* Ec2TaskDefinition/FargateTaskDefinition - To create a Task Definition for the container to start
+* SQSQueue - The queue that the worker is processing from
+* Ec2Service/FargateService - The Service running the container
+
+
+# Part 2: Create `Ec2QueueWorkerServiceProps` interface  and `Ec2QueueWorkerService` construct  
+
+The `Ec2QueueWorkerServiceProps` interface will contain properties to construct the Ec2TaskDefinition, SQSQueue and Ec2Service:
+
+```ts
+/**
+ * Properties to define an ECS service
+ */
+export interface Ec2QueueWorkerServiceProps {
+  /**
+   * The minimum number of CPU units to reserve for the container.
+   *
+   * @default none
+   */
+  readonly cpu?: number;
+
+  /**
+   * The hard limit (in MiB) of memory to present to the container.
+   *
+   * If your container attempts to exceed the allocated memory, the container
+   * is terminated.
+   *
+   * At least one of memoryLimitMiB and memoryReservationMiB is required for non-Fargate services.
+   */
+  readonly memoryLimitMiB?: number;
+
+  /**
+   * The soft limit (in MiB) of memory to reserve for the container.
+   *
+   * When system memory is under contention, Docker attempts to keep the
+   * container memory within the limit. If the container requires more memory,
+   * it can consume up to the value specified by the Memory property or all of
+   * the available memory on the container instance—whichever comes first.
+   *
+   * At least one of memoryLimitMiB and memoryReservationMiB is required for non-Fargate services.
+   */
+  readonly memoryReservationMiB?: number;
+}
+```
+
+An example use case:
+```ts
+// Create the vpc and cluster used by the scheduled task
+const vpc = new ec2.VpcNetwork(stack, 'Vpc', { maxAZs: 1 });
+const cluster = new ecs.Cluster(stack, 'EcsCluster', { vpc });
+cluster.addCapacity('DefaultAutoScalingGroup', {
+  instanceType: new ec2.InstanceType('t2.micro')
+});
+
+// Create the scheduled task
```

Comment:
this does not appear to be a scheduled task?

Reply:
Updated